### PR TITLE
[Documentation] Add multi-task learning CLI notebook example

### PR DESCRIPTION
**Pull request by classicsong**
Add yaml configuration file for model training

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+   "source": [
+    "In this notebook, we will train a RGCN model on the ACM dataset with two training supervisions, i.e., link prediction and node feature reconstruction."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3087d88f",
+   "metadata": {},
+   "source": [
+    "**Note:** More details about multi-task learning please refer to [Multi-task Learning in GraphStorm](https://graphstorm.readthedocs.io/en/latest/advanced/multi-task-learning.html)"
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:32]:
```suggestion
    "**Note:** For more details about multi-task learning, please refer to [Multi-task Learning in GraphStorm](https://graphstorm.readthedocs.io/en/latest/advanced/multi-task-learning.html)"
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a0c7764a",
+   "metadata": {},
+   "source": [
+    "### 0. Setup environment\n",
+    "----\n",
+    "First let's install GraphStorm and its dependencies, PyTorch and DGL"
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:42]:
```suggestion
    "First let's install GraphStorm and its dependencies, PyTorch and DGL."
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "76f8f9c9",
+   "metadata": {},
+   "source": [
+    "### 1. Create the example ACM graph data\n",
+    "---\n",
+    "This notebook uses the ACM graph in the example. We use the following script to create the ACM graph data."
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:68]:
```suggestion
    "This notebook uses the ACM graph as an example. We use the following script to create the ACM graph data."
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    "!wget -O /example/acm_data.py https://github.com/awslabs/graphstorm/raw/main/examples/acm_data.py\n",
+    "!python /example/acm_data.py --output-path /example/acm_raw"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e524e0c6",
+   "metadata": {},
+   "source": [
+    "The ACM graph data includes node files and edge files. It also includes a JSON configuration file describing how to construct a graph for model training. More details can be found in [Use Your Own Data (ACM data example)](https://graphstorm.readthedocs.io/en/latest/tutorials/own-data.html#use-your-own-data)"
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:88]:
```suggestion
    "The ACM graph data includes node files and edge files. It also includes a JSON configuration file describing how to construct a graph for model training. More details can be found in [Use Your Own Data (ACM data example)](https://graphstorm.readthedocs.io/en/latest/tutorials/own-data.html#use-your-own-data)."
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5735fdf1",
+   "metadata": {},
+   "source": [
+    "### 3. GNN Model Training \n",
+    "---\n",
+    "\n",
+    "Once the graph constucted, we can call GraphStorm multi-task learning CLI to run model training. Before kicking off the model training, we need to create a YAML configuration file for the CLI."
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:141]:
```suggestion
    "Once the graph constucted, we can call GraphStorm multi-task learning CLIs to run model training. Before kicking off the model training, we need to create a YAML configuration file for the CLI."
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "994ea961",
+   "metadata": {},
+   "source": [
+    "The YAML configuration file defines two training tasks: \n",
+    " * A link prediction task on the `<paper, citing, paper>` edges. The task specific settings are under the`gsf::multi_task_learning::link_prediction` configuration block.\n",
+    " * A node feature reconstruction task on the `paper` nodes with the node feature `label` to be reconstructed. The task specific settings are under the`gsf::multi_task_learning::reconstruct_node_feat` configuration block.\n",
+    " \n",
+    "For more details of multi-task YAML configuration, please refer to [Define Multi-task for training](https://graphstorm.readthedocs.io/en/latest/advanced/multi-task-learning.html#define-multi-task-for-training)"
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:173]:
```suggestion
    "For more details of multi-task YAML configuration, please refer to the [Define Multi-task for training document](https://graphstorm.readthedocs.io/en/latest/advanced/multi-task-learning.html#define-multi-task-for-training)."
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e7d3182a",
+   "metadata": {},
+   "source": [
+    "#### Launch the training command"
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:181]:
```suggestion
    "#### Launch the training CLI"
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    "           --cf /example/acm_mt.yaml \\\n",
+    "           --num-epochs 4"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e20bce57",
+   "metadata": {},
+   "source": [
+    "The saved model is under `/example/acm_lp/models/`"
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:204]:
```suggestion
    "The saved model is under `/example/acm_lp/models/`."
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+   "source": [
+    "!ls -a /example/acm_lp/models/"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8b1ca0af",
+   "metadata": {},
+   "source": [
+    "### 3. GNN Model Inference \n",
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:222]:
```suggestion
    "### 4. GNN Model Inference \n",
```

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+    "---\n",
+    "\n",
+    "Once the model is trained, we can do model inference with the trained model artifacts with GraphStorm multi-task learning CLI. We can use the same YAML configuration file for model inference."
```
Comment (zhjwy9343) [inline docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:225]:
```suggestion
    "Once the model is trained, we can do model inference with the trained model artifacts by using the GraphStorm multi-task learning CLI. We can use the same YAML configuration file for model inference."
```

**Comment by zhjwy9343** [comment]
Do we need to explain a little of the `gs_multi_task_learning` command?

File: docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb (hunk 5 above)
Comment (classicsong) [reply docs/source/cli/notebooks/Notebook_4_MT_CLI.ipynb:141]: This is a single CLI command. I changed it into `call the GraphStorm multi-task learning CLI to run`

**Comment by classicsong** [comment]
> Do we need to explain a little of the `gs_multi_task_learning` command?

No, it is explained in https://graphstorm.readthedocs.io/en/latest/advanced/multi-task-learning.html#define-multi-task-for-training.

**Review by zhjwy9343 — APPROVED** [review]
LGTM, except for one unsolved comment.

**Comment by classicsong** [comment]
> LGTM, except for one unsolved comment.

I have updated the title to "Launch the trianing job"

**Pull request merged**